include/mbedtls/pk_info.h

```diff
@@ -106,6 +106,14 @@ struct mbedtls_pk_info_t
      * usage restrictions into account. */
     int (*can_do)( const void * ctx, mbedtls_pk_type_t type );
 
+    /** Signature size
+     *
+     * mbedtls_pk_signature_size() returns this value.
+     *
+     * Opaque implementations may omit this method if they do not support
```

Comment:
Question: what's the motivation for making this mandatory? (That's an honest question, I'm happy both ways, just wondering why it was changed.)

Reply:
I hesitated, but decided that it was dangerous to leave it out and cheap to include it. (To minimize code size at the expense of RAM usage, an engine can return a constant which is the largest possible signature size for any key size.) If the engine doesn't provide this function by provides `mbedtls_pk_sign`, then it's up to the user and the engine to agree on the buffer size and no check can be made.